moduleframework/common.py

```diff
@@ -580,7 +586,23 @@ def getIPaddr(self):
 
         :return: str
         """
-        return self.ipaddr
+        return self._ipaddr
+
+    @property
+    def ipaddr(self):
+        """
+        Return protocol (IP or IPv6) address on a guest machine.
+
+        In many cases it should be same as a host machine's and a port
+        should be forwarded to a host.
+
+        :return: str
+        """
+        return self._ipaddr
+
+    @ipaddr.setter
```

Comment:
I'm not sure with this, do you think that ipaddr should be changed from outside? it should be set when artifacts starts, after network is initialized. not from outside.

Reply:
@jscotka `self.ipaddr = <something>` sets the ipaddress. `self._ipaddr` should not be readable directly.